src/main/scala/sigmastate/serialization/ErgoTreeSerializer.scala

```diff
@@ -0,0 +1,65 @@
+package sigmastate.serialization
+
+import sigmastate.SType
+import sigmastate.Values.{Constant, Value}
+import sigmastate.eval.IRContext
+import sigmastate.lang.DeserializationSigmaBuilder
+import sigmastate.utils.Extensions._
+
+import scala.collection.mutable
+
+class ErgoTreeSerializer(IR: IRContext) {
```

Comment:
@greenhat why we need IR?

Reply:
@ergomorphic for constant extraction/injection using `buildTree`. Fixed  (extracted this feature).